crates/telio-network-monitors/src/local_interfaces.rs

```diff
@@ -0,0 +1,93 @@
+use ipnet::Ipv4Net;
+use std::net::{IpAddr, Ipv4Addr};
+
+/// Function type of a function returning interface addresses from the system
+pub type GetIfAddrs = fn() -> std::io::Result<Vec<if_addrs::Interface>>;
+
+/// Function returning interface addresses from the system
+pub fn system_get_if_addr() -> std::io::Result<Vec<if_addrs::Interface>> {
+    if_addrs::get_if_addrs()
+}
+
+/// Method that returns vector of Interfaces on the system
+/// Filtering out meshnet and loopback IP
+pub fn gather_local_interfaces(
```

Comment:
This fn can be part of the `monitor` module. Then you can use traits and define a default behaviour where `if_addrs::get_if_addrs()` is used, so that this crate users don't have to worry about which get_if_addrs fn implementation will be used. And then on tests you can mock and provide your custom implementation with the return values you need..